### PR TITLE
Adding new roughness boundary conditions for SST

### PR DESCRIPTION
**Pull request by marcosch27**
## Proposed Changes
Hello everyone,

This PR is one of two contributions resulting from my Master’s Thesis.

In this PR, I’ve added new roughness boundary conditions for the SST turbulence model, based on the approach described in [this paper](https://onera.hal.science/hal-01071701/document) by Aupoix.

[The attached PDF](https://github.com/user-attachments/files/22388399/SU2_Pull_RoughBCs.pdf) includes documentation as well as validation results for a fully turbulent flat plate test case.

Please let me know if any modifications are needed.

## PR Checklist

- [x] I am submitting my contribution to the develop branch.
- [x] My contribution generates no new compiler warnings (try with --warnlevel=3 when using meson).
- [x] My contribution is commented and consistent with SU2 style (https://su2code.github.io/docs_v7/Style-Guide/).
- [ ] I used the pre-commit hook to prevent dirty commits and used `pre-commit run --all` to format old commits.
- [ ] I have added a test case that demonstrates my contribution, if necessary.
- [ ] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp), if necessary.

### CODE REVIEW

File: Common/include/option_structure.hpp [hunk 1]
```diff
@@ -1101,6 +1101,59 @@ inline SST_ParsedOptions ParseSSTOptions(const SST_OPTIONS *SST_Options, unsigne
   return SSTParsedOptions;
 }
 
+/*!
+ * \brief SST rough-wall boundary conditions Options
+ */
+enum class ROUGHSST_OPTIONS {
+  NONE,                 /*!< \brief No option / default. */
+  WILCOX1998,           /*!< \brief Wilcox 1998 boundary conditions for rough walls  / default. */
```
Comment (pcarruscag) [inline Common/include/option_structure.hpp]: Both are default?

File: Common/include/option_structure.hpp [hunk 2]
```diff
@@ -1101,6 +1101,59 @@ inline SST_ParsedOptions ParseSSTOptions(const SST_OPTIONS *SST_Options, unsigne
...
+  ROUGH_SST_ParsedOptions ROUGHSSTParsedOptions;
+
+  auto IsPresent = [&](ROUGHSST_OPTIONS option) {
+    const auto roughsst_options_end = ROUGHSST_Options + nROUGHSST_Options;
+    return std::find(ROUGHSST_Options, roughsst_options_end, option) != roughsst_options_end;
+  };
+
+  ROUGHSSTParsedOptions.wilcox2006 = IsPresent(ROUGHSST_OPTIONS::WILCOX2006);
+  ROUGHSSTParsedOptions.limiter_knopp = IsPresent(ROUGHSST_OPTIONS::LIMITER_KNOPP);
+  ROUGHSSTParsedOptions.limiter_aupoix = IsPresent(ROUGHSST_OPTIONS::LIMITER_AUPOIX);
```
Comment (pcarruscag) [inline Common/include/option_structure.hpp]: you don't check for 1998? Is it valid if all of these are true?

File: SU2_CFD/src/solvers/CTurbSSTSolver.cpp [hunk 3]
```diff
@@ -441,33 +444,61 @@ void CTurbSSTSolver::BC_HeatFlux_Wall(CGeometry *geometry, CSolver **solver_cont
...
+          solution[0] = (FrictionVel*FrictionVel / sqrt(constants[6]))*min(1.0, kPlus / 90.0);
+
+          const su2double kappa = config->GetwallModel_Kappa();
+          su2double beta_1 = constants[4];
+          solution[1] = min( FrictionVel/(sqrt(constants[6])*d0*kappa), 60.0*laminar_viscosity/(density*beta_1*pow(wall_dist,2))); 
+        }
+        /*--- Aupoix eddy viscosity limiter ---*/ 
+        else if (roughsstParsedOptions.limiter_aupoix) {
+          
+          su2double k0Plus = ( 1.0 /sqrt( constants[6])) * tanh(((log((kPlus +EPS ) / 30.0) / log(10.0)) + 1.0 - 1.0*tanh( (kPlus + EPS) / 125.0))*tanh((kPlus + EPS) / 125.0));
```
Comment (pcarruscag) [inline SU2_CFD/src/solvers/CTurbSSTSolver.cpp]: why log / log(10) instead of log10(...)?

File: SU2_CFD/src/solvers/CTurbSSTSolver.cpp [hunk 4]
```diff
@@ -441,33 +444,61 @@ void CTurbSSTSolver::BC_HeatFlux_Wall(CGeometry *geometry, CSolver **solver_cont
...
+          if (kPlus <= 5)
+            S_R = (200/(kPlus+EPS))*(200/(kPlus+EPS));
+          else
+            S_R = 100/(kPlus+EPS) + ((200/(kPlus+EPS))*(200/(kPlus+EPS)) - 100/(kPlus+EPS))*exp(5-kPlus);
+          
+          solution[0] = 0.0;
+          solution[1] = FrictionVel*FrictionVel*S_R/(laminar_viscosity/density);
+        } 
+        /*--- Knopp eddy viscosity limiter ---*/
+        else if (roughsstParsedOptions.limiter_knopp) {
```
Comment (pcarruscag) [inline SU2_CFD/src/solvers/CTurbSSTSolver.cpp]: The options are mutually exclusive so allowing a list of enums in the config is not the most appropriate.
Better to change to a single enum option?
If it ever becomes possible to specify a list, the config syntax will still be compatible.

File: SU2_CFD/src/solvers/CTurbSSTSolver.cpp [hunk 5]
```diff
@@ -441,33 +444,61 @@ void CTurbSSTSolver::BC_HeatFlux_Wall(CGeometry *geometry, CSolver **solver_cont
...
-          S_R = 100/(kPlus+EPS) + ((200/(kPlus+EPS))*(200/(kPlus+EPS)) - 100/(kPlus+EPS))*exp(5-kPlus);
-
-        /*--- Modify the omega to account for a rough wall. ---*/
-        su2double solution[2];
-        solution[0] = 0.0;
-        solution[1] = FrictionVel*FrictionVel*S_R/(laminar_viscosity/density);
+          if (kPlus <= 5)
+            S_R = (200/(kPlus+EPS))*(200/(kPlus+EPS));
+          else
+            S_R = 100/(kPlus+EPS) + ((200/(kPlus+EPS))*(200/(kPlus+EPS)) - 100/(kPlus+EPS))*exp(5-kPlus);
```
Comment (pcarruscag) [inline SU2_CFD/src/solvers/CTurbSSTSolver.cpp]: Use the pow function please
```suggestion
            S_R = pow(200/(kPlus+EPS), 2);
          else
            S_R = 100/(kPlus+EPS) + (pow(200/(kPlus+EPS), 2) - 100/(kPlus+EPS))*exp(5-kPlus);
```

File: SU2_CFD/src/solvers/CTurbSSTSolver.cpp [hunk 6]
```diff
@@ -441,33 +444,61 @@ void CTurbSSTSolver::BC_HeatFlux_Wall(CGeometry *geometry, CSolver **solver_cont
         su2double kPlus = FrictionVel*Roughness_Height*density/laminar_viscosity;
 
         su2double S_R= 0.0;
+ 
+        su2double solution[2];
```
Comment (pcarruscag) [inline SU2_CFD/src/solvers/CTurbSSTSolver.cpp]:
```suggestion
        su2double solution[2] = {};

```

**Review by pcarruscag — COMMENTED** [review]
Add a testcase please, something of verification quality, are there known results from literature for flatplates?

File: Common/include/option_structure.hpp (hunk 1 above)
Comment (marcosch27) [reply Common/include/option_structure.hpp]: I will fix this issue. When surface roughness is used, the Wilcox boundary condition should be the default. To align with the current implementation, I will set WILCOX2006 as the default boundary condition.

File: SU2_CFD/src/solvers/CTurbSSTSolver.cpp (hunk 3 above)
Comment (marcosch27) [reply SU2_CFD/src/solvers/CTurbSSTSolver.cpp]: In the implementation, the current expression is retained to remain consistent with the formulation in the reference paper, even though it would be mathematically equivalent to using a base-10 logarithm.

Comment (pcarruscag) [reply SU2_CFD/src/solvers/CTurbSSTSolver.cpp]: Put a comment if you like, but this is log10, thanks.

**Comment by marcosch27** [comment]
Hi @pcarruscag and @bigfooted,

As part of this pull request, I’ve a validation test case for the new roughness boundary condition in the SST model.
I’d like to open a pull request to add this test case to the su2code/TestCases repository, but I currently don’t have permission to push a new branch there.

Could you please grant me temporary access or let me know the preferred procedure for contributing this test case?
Alternatively, I can open a pull request from my fork if that’s the standard approach.

Thanks a lot for your help!

**Comment by marcosch27** [comment]
Hi!

I’ve opened a pull request at [su2code/TestCases#178](https://github.com/su2code/TestCases/pull/178), where I’ve uploaded the validation test case for the newly implemented boundary conditions.

The results obtained with the different boundary conditions are presented in terms of the log-law profiles and velocity shift, confirming the findings reported [in the literature](https://onera.hal.science/hal-01071701/document).

File: SU2_CFD/src/variables/CTurbSSTVariable.cpp [hunk 7]
```diff
@@ -91,7 +93,13 @@ void CTurbSSTVariable::SetBlendingFunc(unsigned long iPoint, su2double val_visco
     F1(iPoint) = max(F1(iPoint), F3);
   }
 
-  AD::SetPreaccOut(F1(iPoint)); AD::SetPreaccOut(F2(iPoint)); AD::SetPreaccOut(CDkw(iPoint));
+  /*--- SST Limiter ---*/ 
+  
+  su2double limiter = a1*Solution(iPoint,1) / (F2(iPoint)*vorticity_mag + EPS);
+  limiter = min(1.0, limiter);
+  SST_Limiter(iPoint) = limiter;
```
Comment (pcarruscag) [inline SU2_CFD/src/variables/CTurbSSTVariable.cpp]: @marcosch27 can you leave this for another PR please?
We already use a lot of memory, especially for adjoint, so I'm not keen on adding more storage just to output values.

Comment (marcosch27) [reply SU2_CFD/src/variables/CTurbSSTVariable.cpp]: Ok, that's clear, I've removed all the changes related to the SST limiter output.

File: TestCases/rans/flatplate/roughness/bc_wilcox2006/turb_SST_flatplate_roughBCWilcox2006.cfg [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+CONV_NUM_METHOD_TURB= SCALAR_UPWIND
+%
+% Monotonic Upwind Scheme for Conservation Laws (TVD) in the turbulence equations.
+%           Required for 2nd order upwind schemes (NO, YES)
+MUSCL_TURB= NO
+%
+% Slope limiter (VENKATAKRISHNAN, MINMOD)
+SLOPE_LIMITER_TURB= VENKATAKRISHNAN
+%
+% Time discretization (EULER_IMPLICIT)
```
Comment (pcarruscag) [inline TestCases/rans/flatplate/roughness/bc_wilcox2006/turb_SST_flatplate_roughBCWilcox2006.cfg]: Can you cleanup the unused/default options please? Take a look at our other config files to see what I mean.
```suggestion
MUSCL_TURB= NO
```

File: TestCases/rans/flatplate/roughness/bc_wilcox2006/turb_SST_flatplate_roughBCWilcox2006.cfg [hunk 9]
```diff
@@ -0,0 +1,248 @@
...
+% Convective numerical method (JST, LAX-FRIEDRICH, CUSP, ROE, AUSM, HLLC,
+%                              TURKEL_PREC, MSW)
+CONV_NUM_METHOD_FLOW= ROE
+%
+% Monotonic Upwind Scheme for Conservation Laws (TVD) in the flow equations.
+%           Required for 2nd order upwind schemes (NO, YES)
+MUSCL_FLOW= YES
+%
+% Slope limiter (NONE, VENKATAKRISHNAN, VENKATAKRISHNAN_WANG,
+%                BARTH_JESPERSEN, VAN_ALBADA_EDGE)
```
Comment (pcarruscag) [inline TestCases/rans/flatplate/roughness/bc_wilcox2006/turb_SST_flatplate_roughBCWilcox2006.cfg]: We no longer put the lists of options in all configs because then if we remove or add an option everything is out of date.
```suggestion
```

File: TestCases/rans/flatplate/roughness/bc_wilcox2006/turb_SST_flatplate_roughBCWilcox2006.cfg [hunk 10]
```diff
@@ -0,0 +1,248 @@
...
+MG_POST_SMOOTH= ( 2, 2, 2, 2)
+%
+% Jacobi implicit smoothing of the correction
+MG_CORRECTION_SMOOTH= ( 0, 0, 0, 0 )
+%
+% Damping factor for the residual restriction
+MG_DAMP_RESTRICTION= 0.8
+%
+% Damping factor for the correction prolongation
+MG_DAMP_PROLONGATION= 0.8
```
Comment (pcarruscag) [inline TestCases/rans/flatplate/roughness/bc_wilcox2006/turb_SST_flatplate_roughBCWilcox2006.cfg]: None of this is used if the level is 0, etc.

File: TestCases/rans/flatplate/roughness/bc_wilcox2006/turb_SST_flatplate_roughBCWilcox2006.cfg [hunk 11]
```diff
@@ -0,0 +1,248 @@
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%                                                                              %
+% SU2 configuration file                                                       %
+% Case description: Turbulent flow over flat plate with zero pressure gradient %
+% Author: Thomas D. Economon                                                   %
+% Institution: Stanford University                                             %
+% Date: 2011.11.10                                                             %
+% File Version 5.0.0 "Raven"                                                %
```
Comment (pcarruscag) [inline TestCases/rans/flatplate/roughness/bc_wilcox2006/turb_SST_flatplate_roughBCWilcox2006.cfg]: You can probably grab and modify our most recent config for a flatplate to make it easier to cleanup the files

File: SU2_CFD/src/solvers/CTurbSSTSolver.cpp [hunk 12]
```diff
@@ -441,33 +444,55 @@ void CTurbSSTSolver::BC_HeatFlux_Wall(CGeometry *geometry, CSolver **solver_cont
...
+
+          solution[0] = kwall;
+          solution[1] = omegawallPlus*FrictionVel*FrictionVel*density/laminar_viscosity;
+        }
         /*--- Set the solution values and zero the residual ---*/
         nodes->SetSolution_Old(iPoint,solution);
         nodes->SetSolution(iPoint,solution);
         LinSysRes.SetBlock_Zero(iPoint);
-
+
```
Comment (pcarruscag) [inline SU2_CFD/src/solvers/CTurbSSTSolver.cpp]:
```suggestion

```

File: Common/include/option_structure.hpp [hunk 13]
```diff
@@ -1101,6 +1101,58 @@ inline SST_ParsedOptions ParseSSTOptions(const SST_OPTIONS *SST_Options, unsigne
...
+ */
+inline ROUGH_SST_ParsedOptions ParseROUGHSSTOptions(ROUGHSST_MODEL sstbcs_option) {
+    ROUGH_SST_ParsedOptions opts;
+    opts.version = sstbcs_option;
+
+    switch(sstbcs_option) {
+        case ROUGHSST_MODEL::WILCOX1998: opts.wilcox1998 = true; break;
+        case ROUGHSST_MODEL::WILCOX2006: opts.wilcox2006 = true; break;
+        case ROUGHSST_MODEL::LIMITER_KNOPP: opts.limiter_knopp = true; break;
+        case ROUGHSST_MODEL::LIMITER_AUPOIX: opts.limiter_aupoix = true; break;
```
Comment (pcarruscag) [inline Common/include/option_structure.hpp:1148]: Can you add a check that mutually exclusive options are not used together please?
It's either 98 or 06 and one limiter or the other right?
(we do this for other options of this type too)

File: TestCases/serial_regression.py [hunk 14]
```diff
@@ -253,6 +253,31 @@ def main():
...
+    turb_flatplate_sst_roughBCKnopp.test_vals = [10.000000, 0.053020, -3.454853, -0.684543, -0.886080, 2.140376, 1.043068, 4.808919, -0.203494, 0.053645]
+    test_list.append(turb_flatplate_sst_roughBCKnopp)
+
+    # FLAT PLATE, ROUGHNESS BC AUPOIX SST
+    turb_flatplate_sst_roughBCAupoix           = TestCase('turb_sst_flatplate_roughBCAupoix')
+    turb_flatplate_sst_roughBCAupoix.cfg_dir   = "rans/flatplate/roughness/bc_aupoix"
+    turb_flatplate_sst_roughBCAupoix.cfg_file  = "turb_SST_flatplate_roughBCAupoix.cfg"
+    turb_flatplate_sst_roughBCAupoix.test_iter = 10
+    turb_flatplate_sst_roughBCAupoix.test_vals = [10.000000, 0.053252, -3.575414, -0.761810, -0.998912, 2.003238, 0.907276, 4.807309, -0.197354, 0.051349]
+    test_list.append(turb_flatplate_sst_roughBCAupoix)
```
Comment (pcarruscag) [inline TestCases/serial_regression.py]: Move a couple of these tests to parallel_regression, please, so that all actions take more or less the same time.

**Review by pcarruscag — COMMENTED** [review]
Thank you for the tests, I think it's good to go after these minor comments.

File: Common/include/option_structure.hpp (hunk 13 above)
Comment (marcosch27) [reply Common/include/option_structure.hpp:1148]: No, the naming might indeed be confusing, but these are four separate boundary conditions, and only one should be used at a time. With the current implementation, they should all be treated as mutually exclusive options.

Let me know if any further modifications are needed.

Comment (pcarruscag) [reply Common/include/option_structure.hpp:1148]: Yeah see line 1086 of this file for the type of error checking that I'm mentioning.

Comment (marcosch27) [reply Common/include/option_structure.hpp:1148]: Ok, but with the current formulation, only a single enum value is passed in, so it’s not possible for more than one boundary condition to be selected at the same time. If an additional error check is needed, I can revert to the original formulation that used a list.

Comment (pcarruscag) [reply Common/include/option_structure.hpp:1148]: Ah sorry I misunderstood that

File: TestCases/rans/flatplate/roughness/bc_wilcox2006/turb_SST_flatplate_roughBCWilcox2006.cfg [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+% ------------- DIRECT, ADJOINT, AND LINEARIZED PROBLEM DEFINITION ------------%
+%
+SOLVER= RANS
+KIND_TURB_MODEL= SST
+WALL_ROUGHNESS = (wall, 400e-6)
+KIND_ROUGHSST_MODEL = WILCOX2006
+MATH_PROBLEM= DIRECT
+RESTART_SOL= NO
```
Comment (pcarruscag) [inline TestCases/rans/flatplate/roughness/bc_wilcox2006/turb_SST_flatplate_roughBCWilcox2006.cfg]: You are adding restart files to the testcases repo, so the cases should probably be restarted? (which is good to track converged values instead of just a few iterations)
```suggestion
RESTART_SOL= YES
```

File: .github/workflows/regression.yml [hunk 16]
```diff
@@ -211,7 +211,7 @@ jobs:
         uses: docker://ghcr.io/su2code/su2/test-su2:250717-1402
         with:
           # -t <Tutorials-branch> -c <Testcases-branch>
-          args: -b ${{github.ref}} -t develop -c develop -s ${{matrix.testscript}}
+          args: -b ${{github.ref}} -t develop -c feature_turb_flatplate_SST_RoughBCs -s ${{matrix.testscript}}
```
Comment (pcarruscag) [inline .github/workflows/regression.yml]: Remember to revert this to develop

**Review by pcarruscag — APPROVED** [review]

File: TestCases/rans/flatplate/roughness/bc_aupoix/turb_SST_flatplate_roughBCAupoix.cfg [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+% ------------- DIRECT, ADJOINT, AND LINEARIZED PROBLEM DEFINITION ------------%
+%
+SOLVER= RANS
+KIND_TURB_MODEL= SST
+WALL_ROUGHNESS = (wall, 400e-6)
+KIND_ROUGHSST_MODEL = LIMITER_AUPOIX
+MATH_PROBLEM= DIRECT
+RESTART_SOL= NO
```
Comment (pcarruscag) [inline TestCases/rans/flatplate/roughness/bc_aupoix/turb_SST_flatplate_roughBCAupoix.cfg]:
```suggestion
RESTART_SOL= YES
```

File: TestCases/rans/flatplate/roughness/bc_knopp/turb_SST_flatplate_roughBCKnopp.cfg [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+% ------------- DIRECT, ADJOINT, AND LINEARIZED PROBLEM DEFINITION ------------%
+%
+SOLVER= RANS
+KIND_TURB_MODEL= SST
+WALL_ROUGHNESS = (wall, 400e-6)
+KIND_ROUGHSST_MODEL = LIMITER_KNOPP
+MATH_PROBLEM= DIRECT
+RESTART_SOL= NO
```
Comment (pcarruscag) [inline TestCases/rans/flatplate/roughness/bc_knopp/turb_SST_flatplate_roughBCKnopp.cfg]:
```suggestion
RESTART_SOL= YES
```

File: TestCases/rans/flatplate/roughness/bc_aupoix/turb_SST_flatplate_roughBCAupoix.cfg [hunk 19]
```diff
@@ -0,0 +1,109 @@
...
+%
+MESH_FILENAME= mesh_flatplate_turb_137x97.su2
+MESH_FORMAT= SU2
+MESH_OUT_FILENAME= mesh_out.su2
+SOLUTION_FILENAME= solution_flow.dat
+SOLUTION_ADJ_FILENAME= solution_adj.dat
+TABULAR_FORMAT= CSV
+CONV_FILENAME= history
+RESTART_FILENAME= restart_flow.dat
+RESTART_ADJ_FILENAME= restart_adj.dat
```
Comment (pcarruscag) [inline TestCases/rans/flatplate/roughness/bc_aupoix/turb_SST_flatplate_roughBCAupoix.cfg]:
```suggestion
MESH_FILENAME= mesh_flatplate_turb_137x97
MESH_FORMAT= SU2
MESH_OUT_FILENAME= mesh_out
SOLUTION_FILENAME= restart_flow
SOLUTION_ADJ_FILENAME= restart_adj
TABULAR_FORMAT= CSV
CONV_FILENAME= history
RESTART_FILENAME= restart_flow
RESTART_ADJ_FILENAME= restart_adj
```

File: TestCases/rans/flatplate/roughness/bc_knopp/turb_SST_flatplate_roughBCKnopp.cfg [hunk 20]
```diff
@@ -0,0 +1,109 @@
...
+%
+MESH_FILENAME= mesh_flatplate_turb_137x97.su2
+MESH_FORMAT= SU2
+MESH_OUT_FILENAME= mesh_out.su2
+SOLUTION_FILENAME= solution_flow.dat
+SOLUTION_ADJ_FILENAME= solution_adj.dat
+TABULAR_FORMAT= CSV
+CONV_FILENAME= history
+RESTART_FILENAME= restart_flow.dat
+RESTART_ADJ_FILENAME= restart_adj.dat
```
Comment (pcarruscag) [inline TestCases/rans/flatplate/roughness/bc_knopp/turb_SST_flatplate_roughBCKnopp.cfg]:
```suggestion
MESH_FILENAME= mesh_flatplate_turb_137x97
MESH_FORMAT= SU2
MESH_OUT_FILENAME= mesh_out
SOLUTION_FILENAME= restart_flow
SOLUTION_ADJ_FILENAME= restart_adj
TABULAR_FORMAT= CSV
CONV_FILENAME= history
RESTART_FILENAME= restart_flow
RESTART_ADJ_FILENAME= restart_adj
```

File: TestCases/rans/flatplate/roughness/bc_wilcox2006/turb_SST_flatplate_roughBCWilcox2006.cfg [hunk 21]
```diff
@@ -0,0 +1,109 @@
...
+%
+MESH_FILENAME= mesh_flatplate_turb_137x97.su2
+MESH_FORMAT= SU2
+MESH_OUT_FILENAME= mesh_out.su2
+SOLUTION_FILENAME= solution_flow.dat
+SOLUTION_ADJ_FILENAME= solution_adj.dat
+TABULAR_FORMAT= CSV
+CONV_FILENAME= history
+RESTART_FILENAME= restart_flow.dat
+RESTART_ADJ_FILENAME= restart_adj.dat
```
Comment (pcarruscag) [inline TestCases/rans/flatplate/roughness/bc_wilcox2006/turb_SST_flatplate_roughBCWilcox2006.cfg]:
```suggestion
MESH_FILENAME= mesh_flatplate_turb_137x97
MESH_FORMAT= SU2
MESH_OUT_FILENAME= mesh_out
SOLUTION_FILENAME= restart_flow
SOLUTION_ADJ_FILENAME= restart_adj
TABULAR_FORMAT= CSV
CONV_FILENAME= history
RESTART_FILENAME= restart_flow
RESTART_ADJ_FILENAME= restart_adj
```

File: .github/workflows/regression.yml (hunk 16 above)
Comment (pcarruscag) [inline .github/workflows/regression.yml]:
```suggestion
          args: -b ${{github.ref}} -t develop -c develop -s ${{matrix.testscript}}
```

File: .github/workflows/regression.yml [hunk 22]
```diff
@@ -260,7 +260,7 @@ jobs:
         uses: docker://ghcr.io/su2code/su2/test-su2:250717-1402
         with:
           # -t <Tutorials-branch> -c <Testcases-branch>
-          args: -b ${{github.ref}} -t develop -c develop -s ${{matrix.testscript}} -a "--tapetests"
+          args: -b ${{github.ref}} -t develop -c feature_turb_flatplate_SST_RoughBCs -s ${{matrix.testscript}} -a "--tapetests"
```
Comment (pcarruscag) [inline .github/workflows/regression.yml]:
```suggestion
          args: -b ${{github.ref}} -t develop -c develop -s ${{matrix.testscript}} -a "--tapetests"
```

File: .github/workflows/regression.yml [hunk 23]
```diff
@@ -306,7 +306,7 @@ jobs:
         uses: docker://ghcr.io/su2code/su2/test-su2-tsan:250717-1402
         with:
           # -t <Tutorials-branch> -c <Testcases-branch>
-          args: -b ${{github.ref}} -t develop -c develop -s ${{matrix.testscript}} -a "--tsan"
+          args: -b ${{github.ref}} -t develop -c feature_turb_flatplate_SST_RoughBCs -s ${{matrix.testscript}} -a "--tsan"
```
Comment (pcarruscag) [inline .github/workflows/regression.yml]:
```suggestion
          args: -b ${{github.ref}} -t develop -c develop -s ${{matrix.testscript}} -a "--tsan"
```

File: .github/workflows/regression.yml [hunk 24]
```diff
@@ -351,7 +351,7 @@ jobs:
         uses: docker://ghcr.io/su2code/su2/test-su2-asan:250717-1402
         with:
           # -t <Tutorials-branch> -c <Testcases-branch>
-          args: -b ${{github.ref}} -t develop -c develop -s ${{matrix.testscript}} -a "--asan"
+          args: -b ${{github.ref}} -t develop -c feature_turb_flatplate_SST_RoughBCs -s ${{matrix.testscript}} -a "--asan"
```
Comment (pcarruscag) [inline .github/workflows/regression.yml]:
```suggestion
          args: -b ${{github.ref}} -t develop -c develop -s ${{matrix.testscript}} -a "--asan"
```

File: TestCases/serial_regression.py [hunk 25]
```diff
@@ -253,6 +253,14 @@ def main():
     turb_wallfunction_flatplate_sst.test_vals = [-4.036299, -1.916263, -1.821788, 1.443878, -1.579554, 1.521534, 10.000000, -2.351461, 0.030011, 0.002409]
     test_list.append(turb_wallfunction_flatplate_sst)
 
+    # FLAT PLATE, ROUGHNESS BC WILCOX2006 SST
+    turb_flatplate_sst_roughBCWilcox2006           = TestCase('turb_sst_flatplate_roughBCWilcox2006')
+    turb_flatplate_sst_roughBCWilcox2006.cfg_dir   = "rans/flatplate/roughness/bc_wilcox2006"
+    turb_flatplate_sst_roughBCWilcox2006.cfg_file  = "turb_SST_flatplate_roughBCWilcox2006.cfg"
+    turb_flatplate_sst_roughBCWilcox2006.test_iter = 10
+    turb_flatplate_sst_roughBCWilcox2006.test_vals = [10.000000, 0.052109, -3.937983, -1.063721, -1.483116, 1.514608, -1.003919, 4.472050, -0.180308, 0.033500]
```
Comment (pcarruscag) [inline TestCases/serial_regression.py]:
```suggestion
    turb_flatplate_sst_roughBCWilcox2006.test_vals = [-5.117900, -2.534224, -2.904279, 0.381710, -3.100346, 1.180161, -0.188797, 0.004029]
```

File: TestCases/parallel_regression.py [hunk 26]
```diff
@@ -408,6 +408,22 @@ def main():
     turb_flatplate_CC_Sarkar.test_vals = [-1.195053, 2.089306, 1.529063, 5.164703, -3.700917, 8.162921]
     test_list.append(turb_flatplate_CC_Sarkar)
 
+    # FLAT PLATE, ROUGHNESS BC KNOPP SST
+    turb_flatplate_sst_roughBCKnopp           = TestCase('turb_sst_flatplate_roughBCKnopp')
+    turb_flatplate_sst_roughBCKnopp.cfg_dir   = "rans/flatplate/roughness/bc_knopp"
+    turb_flatplate_sst_roughBCKnopp.cfg_file  = "turb_SST_flatplate_roughBCKnopp.cfg"
+    turb_flatplate_sst_roughBCKnopp.test_iter = 10
+    turb_flatplate_sst_roughBCKnopp.test_vals = [10.000000, 0.053020, -3.454853, -0.684543, -0.886080, 2.140376, 1.043068, 4.808919, -0.203494, 0.053645]
```
Comment (pcarruscag) [inline TestCases/parallel_regression.py]:
```suggestion
    turb_flatplate_sst_roughBCKnopp.test_vals = [-5.058634, -2.460850, -2.847064, 0.447200, -2.595042, 1.497149, -0.188079, 0.004571]
```

File: TestCases/parallel_regression.py [hunk 27]
```diff
@@ -408,6 +408,22 @@ def main():
...
+    test_list.append(turb_flatplate_sst_roughBCKnopp)
+
+    # FLAT PLATE, ROUGHNESS BC AUPOIX SST
+    turb_flatplate_sst_roughBCAupoix           = TestCase('turb_sst_flatplate_roughBCAupoix')
+    turb_flatplate_sst_roughBCAupoix.cfg_dir   = "rans/flatplate/roughness/bc_aupoix"
+    turb_flatplate_sst_roughBCAupoix.cfg_file  = "turb_SST_flatplate_roughBCAupoix.cfg"
+    turb_flatplate_sst_roughBCAupoix.test_iter = 10
+    turb_flatplate_sst_roughBCAupoix.test_vals = [10.000000, 0.053252, -3.575414, -0.761810, -0.998912, 2.003238, 0.907276, 4.807309, -0.197354, 0.051349]
```
Comment (pcarruscag) [inline TestCases/parallel_regression.py]:
```suggestion
    turb_flatplate_sst_roughBCAupoix.test_vals = [-5.278097, -2.297701, -2.883899, 0.228298, -1.375945, 3.209449, -0.188736, 0.007201]
```
